lib/gis/mapset_msc.c

```diff
@@ -197,14 +197,14 @@ int make_mapset_element_impl(const char *p_path, const char *p_element,
         p++;
     /* add trailing slash if missing */
     --p;
-    if (*p++ != '/') {
-        *p++ = '/';
+    if (*p++ != HOST_DIRSEP) {
+        *p++ = HOST_DIRSEP;
         *p = 0;
     }
 
     /* now append element, one directory at a time, to path */
     while (1) {
-        if (*element == '/' || *element == 0) {
+        if (*element == HOST_DIRSEP || *element == 0) {
             *p = 0;
             char *msg = NULL;
 
```

Comment:
Somewhere in the lines below one more change is needed to avoid issues like this (see [OSGeo4W CI run log](https://github.com/OSGeo/grass/actions/runs/10955938081/job/30420885251)):

```
+ v.in.ascii --o -z format=standard input=data/random_points.ref output=random_points
ERROR: Unable to make mapset element vector/random_points
 (C:\Users\runneradmin\nc_spm_full_v2alpha2\__vector_v_what_rast3_test.v.what.rast3_fv_az845_621_4592\vector/random_points): No such file or directory
```